packages/flutter/lib/src/services/text_input.dart

```diff
@@ -413,6 +417,48 @@ class TextInputConfiguration {
   /// Defaults to true.
   final bool autocorrect;
 
+  /// {@template flutter.services.textInput.enableSmartDashes}
+  /// Whether to allow the platform to automatically format dashes.
+  ///
+  /// This flag only affects iOS. It sets UITextSmartDashesType in the engine.
+  /// When true, it passes UITextSmartDashesTypeYes, and when false, it passes
+  /// UITextSmartDashesTypeNo.
+  ///
+  /// As an example of what this does, two consecutive hyphen characters will be
+  /// automatically replaced with one en dash, and three consecutive hyphens
+  /// will become one em dash.
+  ///
+  /// Defaults to true, unless obscureText is true, when it defaults to false.
+  /// This is to avoid the problem where password fields receive autoformatted
+  /// characters.
+  ///
+  /// See also:
+  ///  * [enableSmartQuotes]
+  ///  * <https://developer.apple.com/documentation/uikit/uitextinputtraits>
+  /// {@endtemplate}
+  final bool enableSmartDashes;
```

Comment:
The engine implementation seems to toggle between `default` and `no`. Is there a way for users to set the trait to `UITextSmartQuotesType.yes`?

Reply:
Good point, there is no way right now...  I was under the impression that "yes" and "default" were the same, but could that ever change, or is there any other reason to explicitly allow "yes"?  https://developer.apple.com/documentation/uikit/uitextsmartdashestype?language=objc

Reply:
There seems to be a very slight difference between yes and default, but "default" and "no" should cover most use cases:

> This option selectively enables the smart dash behavior based on the keyboard type. For example, smart dashes are disabled for email address and password keyboards.

Reply:
Seems like having an enum rather than a bool would make this possible to extend later, even if we only support two values today.